File: code_eval/cpp/moderate/challenge_15.cpp
// Created by cschalk on 3/30/16.
// Challenge 15 : Endianness
#include <iostream>

int getBit(int val, int bit) {
    return (val & (1 << bit)) >> bit;
}


int main() {
    int test = 86;
    int test1 = getBit(test, 2);
    int test2 = getBit(test, 3);

    if (test1 == test2) {
        std::cout << "BigEndian" << std::endl;
    } else {
        std::cout << "LittleEndian" << std::endl;
    }

    return 0;
}

File: code_eval/java/moderate/Challenge15.java
package moderate;

import java.nio.ByteOrder;

/**
 * Challenge 15 : Endianness
 * Created by cschalk on 3/21/16.
 */
public class Challenge15 {

    public static void main(String[] args) {
       if (java.nio.ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN) {
           System.out.println("LittleEndian");
       } else {
           System.out.println("BigEndian");
       }
    }
}
